lib/oedipus_lex.rb

```diff
@@ -147,7 +147,7 @@ def to_ruby state, predicates, exclusive
 
       cond = if exclusive or not start_state then
                check
-             elsif start_state =~ /^:/ then
+             elsif /^:/.match?(start_state) then
```

Comment:
While looking through the code I saw this optimization. I'm ok with removing it if you do not like it.